File: Python/String/substring/Check If a Word Occurs As a Prefix of Any Word in a Sentence.py
class Solution:
    def isPrefixOfWord(self, sentence: str, searchWord: str) -> int:
        ind =1
        # Split the sentence into words
        words = sentence.split()
        for word in words:
            # Check if the searchWord is a prefix of the word
            if(word.startswith(searchWord)):
                return ind
            ind+=1
        return -1

File: CPP/String/Check If a Word Occurs As a Prefix of Any Word in a Sentence.cpp
class Solution {
public:
    int isPrefixOfWord(string sentence, string searchWord) {
        stringstream s(sentence);
        string word;
        int ind=0;
        while(s>>word){
            ind++;
            if(word.substr(0,searchWord.size())==searchWord){
                return ind;
            }
        }
        return -1;
    }
};
